command/config/config.go

```diff
@@ -456,5 +449,42 @@ func (s *Instance) UnmarshalJSON(data []byte) error {
 	default:
 		return fmt.Errorf("unknown instance type %s", s.Type)
 	}
+	return nil
+}
+
+// For Instance struct
+func (s *Instance) UnmarshalJSON(data []byte) error {
+	type S Instance
+	type T struct {
+		*S
+		Spec json.RawMessage `json:"spec"`
+	}
+	obj := &T{S: (*S)(s)}
+	if err := json.Unmarshal(data, obj); err != nil {
+		return err
+	}
+
+	if err := s.populateSpec(); err != nil {
+		return err
+	}
+
 	return json.Unmarshal(obj.Spec, s.Spec)
 }
+
+type AliasAnkaBuild AnkaBuild
+
+// UnmarshalJSON required for backwards compatibility
+// with the old registry_url field.
+func (m *AnkaBuild) UnmarshalJSON(data []byte) error {
```

Comment:
2 functions with the same name is a little confusing.

Reply:
they need to be the same name because they are implementing the json.Unmarshaler interface for their respective types. When the JSON unmarshaling process encounters these types, it will use their custom UnmarshalJSON functions.